CHANGELOG.md

```diff
@@ -4,6 +4,10 @@ All notable changes to [Earthly](https://github.com/earthly/earthly) will be doc
 
 ## Unreleased
 
+### Changed
```

Comment:
should this be fixed instead? the old behaviour seems rather buggy.
```suggestion
### Fixed
```

I wonder if there's even a need for the feature flagging? I can't think up a case where people wouldn't want the correct value passed through.

Reply:
> should this be fixed instead? the old behaviour seems rather buggy.
I agree, it seems like more a bug fix than a behavior change, I'm good with changing it.
I guess I decided on calling it a change because I wasn't sure if for whatever reason this fix would somehow change expectations for our users.

> I wonder if there's even a need for the feature flagging? I can't think up a case where people wouldn't want the correct value passed through.

Related to the previous point, I just didn't know if there could be some builds that "work" despite this bug.
For example:

```Earthfile

FROM alpine

foo:
   LET my_val=abc
   BUILD +bar --my_arg=$my_val
   
bar:
  ARG my_arg="123"
  RUN echo $my_arg   
```

https://cloud.earthly.dev/builds/76aeae46-969d-40eb-a8f6-b58e9b2ccbad
It results in +foo getting executed twice (my_arg="", my_arg="abc")

Reply:
twice? wow that seems incredibly buggy.

I think we should just fix it (without a feature flag), I can't imagine people needing the old buggy behaviour.